src/compiler/moduleSpecifiers.ts

```diff
@@ -992,7 +992,8 @@ function tryGetModuleNameAsNodeModule({ path, isRedirect }: ModulePath, { getCan
                 // use the actual directory name, so don't look at `packageJsonContent.name` here.
                 const nodeModulesDirectoryName = packageRootPath.substring(parts.topLevelPackageNameIndex + 1);
                 const packageName = getPackageNameFromTypesPackageName(nodeModulesDirectoryName);
-                const conditions = getConditions(options, importMode === ModuleKind.ESNext);
+                const useImportCondition = importMode === ModuleKind.ESNext || (importMode !== undefined ? false : undefined);
+                const conditions = getConditions(options, useImportCondition);

```

Comment:
This bit is duplicated (based on the chagne); should it be a helper somewhere?

Reply:
+1

Reply:
Nice refactor!